src/script/components/LastMessageTracker/LastMessageTracker.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Wire
+ * Copyright (C) 2024 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {HTMLProps, useState, FC, useEffect} from 'react';
+
+import ko from 'knockout';
+
+import {ChevronIcon, IconButton} from '@wireapp/react-ui-kit';
+
+import {Conversation as ConversationEntity} from '../../entity/Conversation';
+import {Message} from '../../entity/message/Message';
+
+export interface MessageVisibility {
+  message: Message;
+  isVisible: boolean;
+}
+
+export interface LastMessageTrackerProps extends HTMLProps<HTMLElement> {
```

Comment:
This should be a separate component as we want to avoid the entire content of `Conversation` to re-render each time messages are scrolled and messages visibility changes